deno_typescript/globals.d.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+// This scopes the `ts` namespace globally, which is where it exists at runtime
+// when building Deno, but the `typescript/lib/typescript.d.ts` is defined as a
+// module
+
+import * as _ts from "typescript";
+
+declare global {
+  namespace ts {
+    export = _ts;
```

Comment:
Where is this file used? 

I wonder if you've seen this hack I have in `deno_typescript/compiler_main.js`
https://github.com/denoland/deno/blob/718c8c4360b5c42d1c0bcd97d7348908d915212e/deno_typescript/compiler_main.js#L141-L144
basically the `typescript.d.ts` which ships with typescript ends in `export = ts;` and for whatever reason `getSourceFile` doesn't like that...

Reply:
I saw it, but didn't touch it.  I suspect it is doing the same thing I am doing.  What the actual file does, is if it isn't loaded as a CommonJS module, it hoists itself into the global as `ts`, the problem is the `.d.ts` file only expresses it as module.  So to remove the hack we need to essentially do this for the runtime TypeScript/JavaScript.  Let me try to fix that all together then.